lib/bolero-engine/src/rng.rs

```diff
@@ -97,32 +97,11 @@ where
 
         let mut state = RngState::new(self.seed, self.max_len, options);
 
-        let mut valid = 0;
-        let mut invalid = 0;
-        while valid < self.iterations {
+        let start_time = Instant::now();
+        while start_time.elapsed() < self.test_time {
+            eprintln!("running, elapsed is {:?}", start_time.elapsed());
             match test.test(&mut state.next_input()) {
-                Ok(true) => {
-                    valid += 1;
-                    continue;
-                }
-                Ok(false) => {
-                    invalid += 1;
-                    if invalid > self.iterations * 2 {
```

Comment:
Removing this is a regression. This is here to prevent implementing unsatisfiable generators and not knowing they're unsatisfiable:

```rust
let g = gen::<u64>().map(|v| false);
check!().with_generator(g).for_each(|_| {});
```

Currently, this harness will panic with this message. After this change, it'll silently accept it.

Reply:
I tried to understand the check and missed that this was the goal — in practice it seems to have validated that at least 1/3 (amortized) of the inputs are valid. And actually bolero counting only valid iterations, that I discovered while making this change, did explain some weird behavior I had seen while trying to debug why some of my generators were taking so long to run :sweat_smile: 

Anyway, I reintroduced this check with the same (1/3) requirements, along with reintroducing the `with_iterations` parameter :)